### PR TITLE
[GraphIndex][Fix] signed and unsigned integer conversion

### PR DESCRIPTION
**Pull request by GaiYu0**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by BarclayII** [comment]
Oops, I already included it in my PR.  Please feel free to check...

**Comment by GaiYu0** [comment]
How soon will your PR be merged?

File: src/graph/graph.cc [hunk 1]
```diff
@@ -20,14 +20,14 @@ Graph::Graph(IdArray src_ids, IdArray dst_ids, IdArray edge_ids, size_t num_node
   CHECK(IsValidIdArray(edge_ids));
   this->AddVertices(num_nodes);
   num_edges_ = src_ids->shape[0];
-  CHECK(num_edges_ == dst_ids->shape[0]) << "vectors in COO must have the same length";
-  CHECK(num_edges_ == edge_ids->shape[0]) << "vectors in COO must have the same length";
+  CHECK(static_cast<long>(num_edges_) == dst_ids->shape[0]) << "vectors in COO must have the same length";
+  CHECK(static_cast<long>(num_edges_) == edge_ids->shape[0]) << "vectors in COO must have the same length";
```
Comment (jermainewang) [inline src/graph/graph.cc]: Use `uint64_t` instead of `long`.

**Comment by zheng-da** [comment]
sorry about the mess I created. clang seems ok with these problems.
are we going to merge this PR?

**Comment by jermainewang** [comment]
@zheng-da we could merge this

**Pull request merged**